crates/types/src/traits/states.rs

```diff
@@ -65,7 +75,7 @@ pub trait State:
     /// This can also be used to rebuild the state for catchup.
     fn from_header(block_header: &Self::BlockHeader) -> Self;
 
-    /// Construct a genesis state.
+    /// Construct a genesis validated state.
     #[must_use]
     fn genesis() -> Self {
```

Comment:
Probably we should get `instance` here too (example: we might pass in some pre-funded accounts to initialize the state with at genesis, for testing purposes)